src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -67,7 +67,35 @@ private JObject GetInvokeResult(byte[] script, Signers signers = null)
             JObject json = new JObject();
             json["script"] = Convert.ToBase64String(script);
             json["state"] = engine.State;
-            json["gasconsumed"] = engine.GasConsumed.ToString();
+            json["gasconsumed"] = new BigDecimal(engine.GasConsumed, NativeContract.GAS.Decimals).ToString();
+            json["exception"] = GetExceptionMessage(engine.FaultException);
+            try
+            {
+                json["stack"] = new JArray(engine.ResultStack.Select(p => p.ToJson()));
+            }
+            catch (InvalidOperationException)
+            {
+                json["stack"] = "error: recursive reference";
+            }
+            if (engine.State != VMState.FAULT)
+            {
+                ProcessInvokeWithWallet(json, signers);
+            }
+            return json;
+        }
+
+        private JObject GetVerificationResult(UInt160 scriptHash, Signers signers = null)
+        {
+            var snapshot = Blockchain.Singleton.GetSnapshot();
+            var contract = snapshot.Contracts.TryGet(scriptHash);
+            var engine = ApplicationEngine.Create(TriggerType.Verification, new Transaction(), snapshot);
+            var context = engine.LoadScript(contract.Script, CallFlags.None, contract.Manifest.Abi.GetMethod("verify").Offset);
```

Comment:
To properly run `verify` we also need to execute `_initialize` as in `VerifyWitnesses`.